### DISCUSSION TITLE
oscilloscope?

### DISCUSSION THREAD

**Discussion by smsavage32** [naudio/NAudio#1085] [General]
Before I dig into this ... 
Is this library suitable for creating an PC Oscilloscope program that displays the sound card input wave forms on the screen?
![ScopeWantedImage](https://github.com/naudio/NAudio/assets/3305971/62367f0f-f839-4747-ae3d-cdc0ceee022f)

**Comment by markheath** [comment]
There's no built-in visualization, but you will be able to access the raw sample data from which you could draw your own display. Some of the NAudio demos show wave-form visualization which may give you some ideas as a starting point
